### PR TITLE
Split JS code into per-scenario files

### PR DESCRIPTION
**Pull request by dodgerblue**
Signed-off-by: Bogdan Purcareata <bpurcareata@fitbit.com>

### CODE REVIEW

File: app/views/cycle.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    document.history.go(-cycleCurrent - 1);
+    document.onbeforeunload = undefined;
+  }
+}
+
+export function init() {
+  console.log("cycle-views start");
+  cycleCurrent = 0;
+  cycleViewsLoaded = [true, false, false];
+  document.location.assign(`cycle${cycleCurrent}.view`).then(updateCycleViews);
```
Comment (Hexxeh) [inline app/views/cycle.js]: Can this call be de-duped with the one in `cycleNext`?

File: app/views/cycle.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  document.onbeforeunload = (evt) => {
+    console.log("onbeforeunload called");
+    evt.preventDefault();
+    document.history.go(-cycleCurrent - 1);
+    document.onbeforeunload = undefined;
+  }
+}
+
+export function init() {
```
Comment (Hexxeh) [inline app/views/cycle.js:64]: You might want to return the promise from these functions, and handle it in the common code, else it'll be silently dropped upon any error. There's no error here, but it'll help whoever starts tweaking this code as a starting point and introduces an error.

File: app/views/session.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+          console.log("resetting view directly");
+          background.x = 0;
+        }
+
+        /* save the old session handling, we'll need it in case session is not finished */
+        sessionBackswipeCallback = document.onbeforeunload;
+
+        /* leave some time for the animation to happen, then load the new view */
+        setTimeout(() => {
+          document.location.assign('session-finish.view').then(updateFinishView);
```
Comment (Hexxeh) [inline app/views/session.js]: Same comment here about silent promise failures, but this one probably needs explicit handling.

**Review by Hexxeh — APPROVED** [review]

File: app/views/cycle.js (hunk 1 above)
Comment (dodgerblue) [reply app/views/cycle.js]: Yep, fixed, will be visible in the next push. Thanks!

File: app/views/cycle.js (hunk 2 above)
Comment (dodgerblue) [reply app/views/cycle.js:64]: Thanks, fixed, will be visible in the next push.

**Review by Hexxeh — APPROVED** [review]

**Review by npclaudiu — APPROVED** [review]

**Pull request merged**